### ISSUE TITLE
Add step definitions needed for enable / disable buttons on Project Setup module 

### ISSUE THREAD

**Issue by aldefouw** [aldefouw/redcap_cypress#109]

**Comment by cdebacker** [comment]
Implemented step definitions for enabling/disabling surveys and longitudinal mode, see commit: https://github.com/cdebacker/redcap_cypress/commit/dbe907d58d051613378502ece6b890a509afaefd

Currently working on step definition for clicking on checkboxes selected by their proximity to matching text, which will be used for configuring project setup within the "Additional customizations" dialog box.
- I click on the checkbox nearest the text {string}

For other configurables, I think it would be best to take a modular approach that involves more explicit Gherkin, like:
1. I click on the button nearest the text "Designate an email field for communciations"
2. I select "email_v2" from the dropdown nearest the text "Choose an email field to use for invitations to survey participants:"

... instead of a "macro" approach:
- I designate the field "email_v2" for email communications

While the first approach isn't always as concise, it eliminates ambiguity of what user input actions are actually being taken, and reduces step definition bloat, which should help Gherkin authors (fewer steps to memorize or search through) and improve code maintainability. Since you mentioned that it's preferable to have to update feature files over step definitions, I think it's best to have the step definitions be more abstract and the Gherkin more concrete in this regard, rather than vice-versa. Let me know what you think about this. It might be worthwhile to create another issue to discuss and document design decisions of this sort, but I'll leave that at your discretion. @aldefouw

**Comment by aldefouw** [comment]
@cdebacker  - I agree with most of your assessment.  

I definitely think the "macro" approach would be a mistake.  If I'm understanding your approach, you'd have to create a separate step definition for every unique field (e.g. "email communications").

One thing I'd like to mention though ... for the dropdowns, I'd like to see an approach of selecting the option by the value that visible to the end user rather than the "value" attribute specified in the option HTML.  

In other words, I'd like to see you select the option by Y rather than x: 
`<option value="x">Y</option>`

Side note ... would your method supersede what Tintin did or would it supplement what he did? 
https://aldefouw.github.io/redcap_cypress/#i-select-string-from-the-dropdown-identified-by-string-labeled-string

(Also noticing that Tintin and Rushi have almost the same step definitions .. so we'll need to consolidate.)

**Comment by cdebacker** [comment]
@aldefouw I believe it would only supplement Tintin's step definition, since mine would achieve the same result but without using selectors, which per my understanding, we are trying to get away from. I say supplement because it might be useful to keep his around as a backup option for development/testing purposes. My goal is to get this approach to work for any situation that isn't already covered by other step definitions, and by adding in "n-th selector" support, it should cover the vast majority of cases (though it might occasionally require some trial and error from test writers, trying different values of n).

For others, we might still need a specific step definition, e.g. if the email communications field dropdown didn't have any nearby text, we could just make an exception by having a step definition 'I designate the field "email_v2" for email communications. If these exceptions become excessive, we can rethink ways to generically identify elements. In the worst case scenario, a simple 'I select <value> from the <nth> dropdown' would work, but I'm thinking we can do better if it becomes necessary.

**Comment by mmcev106** [comment]
Closing.  This is now possible via generic steps like ` I click on the button labeled "Enable" in the row labeled "Clinical Data Pull from EHR"`.
